sdk/program/src/message/compiled_keys.rs

```diff
@@ -49,7 +49,7 @@ impl CompiledKeys {
             }
         }
         if let Some(payer) = &payer {
-            let mut meta = key_meta_map.entry(*payer).or_default();
+            let meta = key_meta_map.entry(*payer).or_default();
             meta.is_signer = true;
             meta.is_writable = true;
```

Comment:
How can this work without a `mut`? :|
Seems like a false positive to me: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=fa3e6986a1fac2feb7bd38496fa5aced
Does CI check compilation errors for `sdk/program`?

This warning found false positives (that were also incorrectly updated) in the other repo that I was just fixing.